service/src/test/java/com/ge/predix/acs/privilege/management/dao/GraphResourceRepositoryTest.java

```diff
@@ -505,29 +506,14 @@ public void testGetResourceEntityAndDescendantsIds() {
     }
 
     @Test
-    public void testGetVersionCreatesOnlyOneVertex() {
+    public void testVersion() {
         assertThat(IteratorUtils.count(this.graph.vertices()), equalTo(0L));
-        int version = this.resourceRepository.getVersion();
+        this.resourceRepository.createVersionVertex(1);
+        assertThat(this.resourceRepository.checkVersionVertexExists(1), equalTo(true));
         assertThat(IteratorUtils.count(this.graph.vertices()), equalTo(1L));
-        assertThat(version, equalTo(0));
-        version = this.resourceRepository.getVersion();
-        version = this.resourceRepository.getVersion();
-        assertThat(IteratorUtils.count(this.graph.vertices()), equalTo(1L));
-        assertThat(version, equalTo(0));
-    }
-
-    @Test
-    public void testSetVersionCreatesOnlyOneVertex() {
-        assertThat(IteratorUtils.count(this.graph.vertices()), equalTo(0L));
-        this.resourceRepository.setVersion(2);
-        int version = this.resourceRepository.getVersion();
-        assertThat(IteratorUtils.count(this.graph.vertices()), equalTo(1L));
-        assertThat(version, equalTo(2));
-        this.resourceRepository.setVersion(3);
-        this.resourceRepository.setVersion(4);
-        assertThat(IteratorUtils.count(this.graph.vertices()), equalTo(1L));
-        version = this.resourceRepository.getVersion();
-        assertThat(version, equalTo(4));
+        
+        this.resourceRepository.createVersionVertex(2);
+        assertThat(IteratorUtils.count(this.graph.vertices()), equalTo(2L));
```

Comment:
@bhuvan-rk this is testing that a new vertex is added. the exists test is being done with version 1.  Though, a exists == false check should be added before version 1 is created.